libs/ledgerjs/packages/hw-app-eth/src/modules/EIP712/EIP712.types.ts

```diff
@@ -23,6 +23,18 @@ export type EIP712MessageTypes = {
   [key: string]: EIP712MessageTypesEntry[];
 };
 
+// As defined in [spec](https://eips.ethereum.org/EIPS/eip-712), the properties below are all required.
+export function isEIP712Message(
```

Comment:
As it is a predicate function I find that it has more sens with the declaration of the type. It has no business logic. It is related to type narrowing.

Reply:
Actually I do agree with @haammar-ledger that it should probably be set in the `utils` rather than in `types`. I know **right now** it's probably being used only in type narrowing for typescript type guards, but it could very well be used for some business logic actually. 

For example: Right now we're inferring that a string that can be parsed into a JSON object should go through the `hw-eth-app/signEIP712Message` method, but it's not necessarly true since an JSON object could be anything and not specifically an EIP712 message. This method could very well be used to be even more specific in your `prepareMessageToSign` and fallback to signing a string _as is_ if it doesn't have the right properties.